### ISSUE TITLE
Possible issue with Adaptivity and Checkpointing

### ISSUE THREAD

**Issue by permcody** [idaholab/moose#11887]
## Rationale

A user has reported that they have received the following error when running a simulation with adaptivity and checkpointing:

```
*** ERROR ***
Can't output Elems with invalid ids!
```
https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/moose-users/M51iTifNb30/skfw0y-LDQAJ

The stack trace shows that MOOSE is in the Checkpoint write inside of StatefulMaterialProperties when this occurs so it seems like something may be getting out of sync or some corruption may be occurring.

## Description

Unfortunately, we don't have a reasonably sized test case that can reproduce this problem and we have a few tests in the framework that tests this combination of capabilities so it may be difficult to reproduce. 

## Impact

Unknown: Possible deficiencies when using checkpointing with adaptivity.

**Comment by friedmud** [comment]
Yep - I'd agree that this is probably a hole in our current capability.  Just the _wrong_ mix of options!  Weird to see a "true" defect!
